### PR TITLE
Login Onboarding: add a survey to the end

### LINKED ISSUE #7405: Login Onboarding: add a survey to the end
Design: https://href.li/?tJpUmVJ2e16EQGeLNSc753-fi-260%3A3942

Screen Shot 2022-08-03 at 2 50 46 PM

### PR DESCRIPTION
**Pull request by jaclync**
Closes: #7405 

#### Why

In order to understand more about the potential motivation for installing the app, we are adding a survey to the end of the login onboarding as an experiment. Since we don't have an A/B testing setup ready, this feature is enabled for all users in release 9.8 even though I still implemented it under a feature flag.

#### Major changes

The survey view was implemented in SwiftUI in `LoginOnboardingSurveyView`, with a survey option enum `LoginOnboardingSurveyOption`. This survey view is shown above the buttons container in `LoginOnboardingViewController` on top of other pre-existing views as a simple way to implement this experimental feature. I thought about refactoring the existing views but there will be more changes than we want for an experiment. When the user taps "Next" on the survey screen when an option is selected, an event `*_login_onboarding_survey_submitted` is logged with the option. When the survey is shown, we also log an event `*_login_onboarding_survey_shown`.

### Testing instructions

- Reinstall the app if you've seen the login onboarding. Alternatively, you can hardcode the onboarding to be always shown by commenting out [these lines](https://github.com/woocommerce/woocommerce-ios/blob/2c2fc77b9d0611c25694fe9bd96989192c320118/WooCommerce/Classes/ViewRelated/AppCoordinator.swift#L140-L143) in `AppCoordinator`
- Tap "Next" three times --> a survey should be shown with the Next button disabled, and an event `*_login_onboarding_survey_shown` should be logged
- Tap on any option --> the selected option should show the color as in the design, and the Next button becomes enabled
- Tap "Next" to continue to the prologue screen --> an event `*_login_onboarding_survey_submitted` with the corresponding `option` value should be logged

### Screenshots

https://user-images.githubusercontent.com/1945542/182781303-8c63492a-f48b-4a49-ad92-9d15b982b9e2.mov

light | dark
-- | --
Simulator Screen Shot - iPhone 13 - 2022-08-04 at 12 09 53 | Simulator Screen Shot - iPhone 13 - 2022-08-04 at 12 10 00

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Authentication/Prologue/LoginOnboardingSurveyView.swift [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    @State private var selectedOption: LoginOnboardingSurveyOption?
+
+    /// - Parameter onSelection: called when an option is selected.
+    init(onSelection: @escaping (LoginOnboardingSurveyOption) -> Void) {
+        self.onSelection = onSelection
+    }
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 52) {
+            Text("What brings you to the WooCommerce app today?")
```
Comment (itsmeichigo) [inline WooCommerce/Classes/Authentication/Prologue/LoginOnboardingSurveyView.swift]: We should localize this copy right?

File: WooCommerce/Classes/Authentication/Prologue/LoginOnboardingSurveyView.swift [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                .secondaryTitleStyle()
+
+            VStack(spacing: 16) {
+                ForEach(options, id: \.self) { option in
+                    // An if/else is used instead of `.buttonStyle(ButtonStyle(isSelected: selectedOption == option))`
+                    // because a weird background color is set to the button briefly after it is selected in the latter case.
```
Comment (itsmeichigo) [inline WooCommerce/Classes/Authentication/Prologue/LoginOnboardingSurveyView.swift]: Thanks for the explanation!

**Review by itsmeichigo — APPROVED** [review]
This works as expected 💯 There's a missing localized string, but I'm approving this anyway so please feel free to merge it after updating the string 😄

File: WooCommerce/Classes/Authentication/Prologue/LoginOnboardingSurveyView.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/Authentication/Prologue/LoginOnboardingSurveyView.swift]: Totally my bad 🤦🏻‍♀️ thanks for catching this! updated in https://github.com/woocommerce/woocommerce-ios/pull/7416/commits/1b61579c168f8828fa6944bc01be63e34862ca14

**Comment by jaclync** [comment]
Thank you for catching the localization issue 🙌 merging now!

**Pull request merged**